src/benchmarks/micro/libraries/System.Text.Json/Serializer/WriteJson.cs

```diff
@@ -67,5 +82,42 @@ public async Task SerializeToStream()
 
         [Benchmark]
         public string SerializeObjectProperty() => JsonSerializer.Serialize(_objectWithObjectProperty);
+
+        private sealed class ArrayBufferWriter : IBufferWriter<byte>
```

Comment:
have we considered adding such type to BCL? it's not the first time I see us re-implementing it

Reply:
I don't think this would be a good implementation for BCL. A canonical IBufferWriter would avoid resizing and use chunks from a buffer pool.